src/transactions/types.ts

```diff
@@ -261,38 +239,29 @@ export type InputSimulateTransactionOptions = {
  */
 export interface InputGenerateSingleSignerRawTransactionData {
   sender: AccountAddressInput;
-  options?: InputGenerateTransactionOptions;
   data: InputGenerateTransactionPayloadData;
-}
-
-/**
- * Interface that holds the user data input when generating a fee payer transaction
- */
-export interface InputGenerateFeePayerRawTransactionData {
-  sender: AccountAddressInput;
-  hasFeePayer: true;
-  secondarySignerAddresses?: AccountAddressInput[];
   options?: InputGenerateTransactionOptions;
-  data: InputGenerateTransactionPayloadData;
+  withFeePayer?: boolean;
+  secondarySignerAddresses?: undefined;
```

Comment:
typescript doesnt know how to solve the return type without having the same type props in all function overload. 

@gregnazario not sure how it worked for you before. I think it didnt really infer the return type because we also use the general type in the function overload https://github.com/aptos-labs/aptos-ts-sdk/blob/main/src/api/transactionSubmission.ts#L47

Reply:
Oh, I see without the overload you'd have to do splitting of the output type.

Does it matter though?

Reply:
it matters since we want to return to the user the correct type so they wont need to
```
if (transaction typeof SingleSignerTransaction){....}
```